### DISCUSSION TITLE
Customize kiali-controlplane role

### DISCUSSION THREAD

**Discussion by HSoulat** [kiali/kiali#5881] [Q&A]
On VmWare Tanzu clusters we need to customize kiali-controlplane role in order to add specific permissions. Unfortunately there is no opion in Kiali CR to customize it.

**Comment by aljesusg** [comment]
Hi @HSoulat, Thanks for the feedback.

Kiali support [Namespace access control](https://kiali.io/docs/configuration/rbac/).
Can you give us more information about the use case? What kind of specific permissions?

Thanks !

**Comment by jmazzitelli** [comment]
Converting to a Q&A discussion.

Yes, there is no way to modify the control plane role (in fact, there is not much in the way of changing the normal roles, too).

We do not want to allow users to modify the roles for obvious security reasons and for reasons that Kiali will break if modified incorrectly.

We would need to hear a compelling use-case to change our installation mechanisms to allow someone to go in and modify the roles.

What you can do is modify it yourself (e.g. you can generate the yaml via `helm template` and then modify it however you wish and then apply the template yaml). Of course, doing this would require you are willing to take the security risk that doing so may cause problems (not to mention the fact you might break Kiali :)

**Comment by HSoulat** [comment]
[VmWare Tanzu](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-releases/services/rn/vmware-tanzu-kubernetes-releases-release-notes/index.html) are following K8s release with delay. Unfortunately they are not yet at K8s 1.25+ version and they still use and enforce PodSecurityPolicies. 
As you said on [issue #1294](https://github.com/kiali/kiali/issues/1294#issue-475046926) there is no reason to support it on Kiali as the feature is depreciated. However the only blocking point is with that is on the control plane role.

Sure, we can generate/modify the kiali-server template in order to customize the control plane role, but in this case we will be out of Kiali standard deployements.
